code/game/objects/items/tools/welder.dm

```diff
@@ -138,6 +138,14 @@
 	remove_fuel(amount)
 	return TRUE
 
+/obj/item/weldingtool/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	if(!tool_enabled)
+		return
+	else
+		remove_fuel(0.5)
+		return
```

Comment:
You can simplify this by removing the else and return
```suggestion
	remove_fuel(0.5)
```

Reply:
committed